src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -60,6 +64,32 @@ namespace {
 BootReasonType DetermineBootReason()
 {
 #ifdef CONFIG_HWINFO
+
+#ifdef CONFIG_SOC_SERIES_NRF54HX
+    uint32_t reason = nrf_resetinfo_resetreas_global_get(NRF_RESETINFO);
+
+    if (reason & RESETINFO_RESETREAS_GLOBAL_RESETPORONLY_Msk)
+    {
+        return BootReasonType::kBrownOutReset;
+    }
+
+    if (reason & RESETINFO_RESETREAS_GLOBAL_DOG_Msk)
+    {
+        return BootReasonType::kHardwareWatchdogReset;
+    }
+
+    if ((reason & (RESETINFO_RESETREAS_GLOBAL_RESETPIN_Msk | RESETINFO_RESETREAS_GLOBAL_RESETPOR_Msk)) ==
+        (RESETINFO_RESETREAS_GLOBAL_RESETPIN_Msk | RESETINFO_RESETREAS_GLOBAL_RESETPOR_Msk))
+    {
+        return BootReasonType::kPowerOnReboot;
+    }
+
+    if ((reason & (RESETINFO_RESETREAS_GLOBAL_RESETPOR_Msk | RESETINFO_RESETREAS_GLOBAL_SECSREQ_Msk)) ==
+        (RESETINFO_RESETREAS_GLOBAL_RESETPOR_Msk | RESETINFO_RESETREAS_GLOBAL_SECSREQ_Msk))
+    {
+        return BootReasonType::kSoftwareUpdateCompleted;
+    }
```

Comment:
Don't you need to clear the reset cause somewhere so that it is not accumulated in subsequent reboots?